src/network/dns/client.rs

```diff
@@ -32,12 +31,22 @@ impl Client {
 
         let handle = Builder::new().name("resolver".into()).spawn({
             let cache = cache.clone();
-            let pending = pending.clone();
             move || {
                 runtime.block_on(async {
                     let resolver = Arc::new(resolver);
 
                     while let Some(ips) = rx.recv().await {
+                        let ips = ips
+                            .into_iter()
+                            .filter(|ip| {
+                                let mut pending = pending.lock().unwrap().clone();
```

Comment:
Do you think we can move the mutex lock outside of the loop? Otherwise it'll constantly be locking/unlocking for every ip in the list. Or is there a consideration here I'm not seeing?

Reply:
Good idea. Will do.